lib/amazonka-core/src/Amazonka/Sign/V4.hs

```diff
@@ -42,7 +43,16 @@ presign ex rq a r ts = Base.signRequest meta mempty auth
         . pair (CI.original hAMZSignedHeaders) (toBS shs)
         . pair (CI.original hAMZToken) (toBS <$> _authSessionToken a)
 
-    digest = Base.Tag "UNSIGNED-PAYLOAD"
+    digest =
+      case _requestBody rq of
```

Comment:
Can you please document why we're splitting the things the way we're doing (e.g., by paraphrasing my comment in #766)?

Reply:
Done. See the haddock of the function.